var/spack/repos/builtin/packages/process-in-process/package.py

```diff
@@ -27,157 +37,167 @@ class ProcessInProcess(Package):
-        when="@3 os=centos7",
-    )
-    resource(
-        name="PiP-glibc",
-        git="https://github.com/RIKEN-SysSoft/PiP-glibc.git",
-        branch="centos/glibc-2.17-260.el7.pip.branch",
-        destination="PiP-glibc",
-        when="@3 os=rhel7",
-    )
-    #   for rhel/centos 8
-    resource(
-        name="PiP-glibc",
-        git="https://github.com/RIKEN-SysSoft/PiP-glibc.git",
-        branch="centos/glibc-2.28-72.el8_1.1.pip.branch",
-        destination="PiP-glibc",
-        when="@3 os=centos8",
-    )
-    resource(
-        name="PiP-glibc",
-        git="https://github.com/RIKEN-SysSoft/PiP-glibc.git",
-        branch="centos/glibc-2.28-72.el8_1.1.pip.branch",
-        destination="PiP-glibc",
-        when="@3 os=rhel8",
-    )
-
-    #  PiP-gdb resource
-    #   for rhel/centos 7
-    resource(
-        name="PiP-gdb",
-        git="https://github.com/RIKEN-SysSoft/PiP-gdb.git",
-        branch="centos/gdb-7.6.1-94.el7.pip.branch",
-        destination="PiP-gdb",
-        when="@3 os=centos7",
-    )
-    resource(
-        name="PiP-gdb",
-        git="https://github.com/RIKEN-SysSoft/PiP-gdb.git",
-        branch="centos/gdb-7.6.1-94.el7.pip.branch",
-        destination="PiP-gdb",
-        when="@3 os=rhel7",
-    )
-    #   for rhel/centos 8
-    resource(
-        name="PiP-gdb",
-        git="https://github.com/RIKEN-SysSoft/PiP-gdb.git",
-        branch="centos/gdb-8.2-12.el8.pip.branch",
-        destination="PiP-gdb",
-        when="@3 os=centos8",
-    )
-    resource(
-        name="PiP-gdb",
-        git="https://github.com/RIKEN-SysSoft/PiP-gdb.git",
-        branch="centos/gdb-8.2-12.el8.pip.branch",
-        destination="PiP-gdb",
-        when="@3 os=rhel8",
-    )
+    if enable_pip_gdb:
+        #  PiP-gdb resource
+        #   for rhel/centos 7
+        resource(
+            name="PiP-gdb",
+            git="https://github.com/procinproc/PiP-gdb.git",
+            branch="centos/gdb-7.6.1-94.el7.pip.branch",
+            destination="PiP-gdb",
+            when="@2 os=centos7",
+        )
+        resource(
+            name="PiP-gdb",
+            git="https://github.com/procinproc/PiP-gdb.git",
+            branch="centos/gdb-7.6.1-94.el7.pip.branch",
+            destination="PiP-gdb",
+            when="@2 os=rhel7",
+        )
+        #   for rhel/centos 8
+        resource(
+            name="PiP-gdb",
+            git="https://github.com/procinproc/PiP-gdb.git",
+            branch="centos/gdb-8.2-12.el8.pip.branch",
+            destination="PiP-gdb",
+            when="@2 os=centos8",
+        )
+        resource(
+            name="PiP-gdb",
+            git="https://github.com/procinproc/PiP-gdb.git",
+            branch="centos/gdb-8.2-12.el8.pip.branch",
+            destination="PiP-gdb",
+            when="@2 os=rhel8",
+        )
+
+    if enable_pip_v3:
+        # resources for PiP version 3
+        #  PiP-glibc resource
+        #   for rhel/centos 7
```

Comment:
Why are we protecting these directives under a class-level conditional?

Reply:
Currently PiP v3 is not stable and not recommended for users to install this version. I will be working on this and it should be stable in the future. This is the reason to have this conditional flag.

Reply:
Can't we still use a variant (or any other mechanism), and put a note on instability of PiP v3? Note that we also have [sticky variants](https://spack.readthedocs.io/en/latest/packaging_guide.html#sticky-variants), whose default value needs to be changed explicitly by users to take effect. That might prevent `clingo` accidentally selecting PiP v3, if that is what concerns you.

Reply:
OK, I have pushed new one using variants.